apps/condo/migrations/20220704172756-0143_ticket_completedat_ticketchange_completedatfrom_and_more.js

```diff
@@ -0,0 +1,65 @@
+// auto generated by kmigrator
+// KMIGRATOR:0143_ticket_completedat_ticketchange_completedatfrom_and_more:IyBHZW5lcmF0ZWQgYnkgRGphbmdvIDQuMCBvbiAyMDIyLTA3LTA0IDEyOjI3Cgpmcm9tIGRqYW5nby5kYiBpbXBvcnQgbWlncmF0aW9ucywgbW9kZWxzCgoKY2xhc3MgTWlncmF0aW9uKG1pZ3JhdGlvbnMuTWlncmF0aW9uKToKCiAgICBkZXBlbmRlbmNpZXMgPSBbCiAgICAgICAgKCdfZGphbmdvX3NjaGVtYScsICcwMTQyX3BheW1lbnRfb3JkZXJfcGF5bWVudGhpc3RvcnlyZWNvcmRfb3JkZXInKSwKICAgIF0KCiAgICBvcGVyYXRpb25zID0gWwogICAgICAgIG1pZ3JhdGlvbnMuQWRkRmllbGQoCiAgICAgICAgICAgIG1vZGVsX25hbWU9J3RpY2tldCcsCiAgICAgICAgICAgIG5hbWU9J2NvbXBsZXRlZEF0JywKICAgICAgICAgICAgZmllbGQ9bW9kZWxzLkRhdGVUaW1lRmllbGQoYmxhbms9VHJ1ZSwgbnVsbD1UcnVlKSwKICAgICAgICApLAogICAgICAgIG1pZ3JhdGlvbnMuQWRkRmllbGQoCiAgICAgICAgICAgIG1vZGVsX25hbWU9J3RpY2tldGNoYW5nZScsCiAgICAgICAgICAgIG5hbWU9J2NvbXBsZXRlZEF0RnJvbScsCiAgICAgICAgICAgIGZpZWxkPW1vZGVscy5EYXRlVGltZUZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSd0aWNrZXRjaGFuZ2UnLAogICAgICAgICAgICBuYW1lPSdjb21wbGV0ZWRBdFRvJywKICAgICAgICAgICAgZmllbGQ9bW9kZWxzLkRhdGVUaW1lRmllbGQoYmxhbms9VHJ1ZSwgbnVsbD1UcnVlKSwKICAgICAgICApLAogICAgICAgIG1pZ3JhdGlvbnMuQWRkRmllbGQoCiAgICAgICAgICAgIG1vZGVsX25hbWU9J3RpY2tldGhpc3RvcnlyZWNvcmQnLAogICAgICAgICAgICBuYW1lPSdjb21wbGV0ZWRBdCcsCiAgICAgICAgICAgIGZpZWxkPW1vZGVscy5EYXRlVGltZUZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgIF0K
+
+exports.up = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Add field completedAt to ticket
+--
+ALTER TABLE "Ticket" ADD COLUMN "completedAt" timestamp with time zone NULL;
+--
+-- Add field completedAtFrom to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "completedAtFrom" timestamp with time zone NULL;
+--
+-- Add field completedAtTo to ticketchange
+--
+ALTER TABLE "TicketChange" ADD COLUMN "completedAtTo" timestamp with time zone NULL;
+--
+-- Add field completedAt to tickethistoryrecord
+--
+ALTER TABLE "TicketHistoryRecord" ADD COLUMN "completedAt" timestamp with time zone NULL;
+--
+-- Fill tickets completedAt field
+--
+UPDATE "Ticket"
+SET "completedAt" = "TicketCompletedTime"."completedAt"
+FROM (
```

Comment:
This inner query looks like not connected to a ticket, we are going to update. It seems like it will select maximal createdAt value among TicketChange record, connected to some Ticket record, which by order will be out of explicit control.
Please, double check this request on a large amount of data.